prog/analysis.go

```diff
@@ -76,7 +76,7 @@ func (s *state) analyzeImpl(c *Call, resources bool) {
 			}
 		case *BufferType:
 			a := arg.(*DataArg)
-			if a.Dir() != DirOut && len(a.Data()) != 0 {
+			if a.Dir() != DirOut && len(a.Data()) != 0 && !typ.IsCompressed() {
```

Comment:
This is not needed for correctness, but is good for performance since w/o this check we will convert the []byte to string (which involves memory allocation and data copy).
However, this is not about compressed/not-compressed. We also don't need the string conversion for BufferBlob and other types. We only need to enter this block of code for BufferString/BufferFilename that we handle below. So I would check these types here, it will also improve performance for BufferBlob.